# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/base/PresShell.cpp
# Commit: 8079db41c325
# Full Hash: 8079db41c32526a9535056ce616c5383dc2a56c1
# Author: Timothy Nikkel <tnikkel@gmail.com>
# Date: 2022-08-29 21:45:26
# Description:
#   Bug 1787684. Only return views with widgets from FindFloatingViewContaining. r=emilio
#   
#   The code that bug 1754436 deleted checked to make sure there was a widget for the returned view, but it looks like it's possible in some edge cases to have an open popup that doesn't have a widget (most likely just for a split second). We need a widget to dispatch the event, and if it doesn't have a widget it's not visible so no point in touching it at all anyways.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155819
# ==============================================================================

diff -r 251d145c0b4e -r 8079db41c325 layout/base/PresShell.cpp
--- a/layout/base/PresShell.cpp	Mon Aug 29 09:46:13 2022 +0000
+++ b/layout/base/PresShell.cpp	Mon Aug 29 09:58:11 2022 +0000
@@ -5642,8 +5642,9 @@
 static nsView* FindFloatingViewContaining(nsPresContext* aRootPresContext,
                                           nsIWidget* aRootWidget,
                                           const LayoutDeviceIntPoint& aPt) {
-  nsIFrame* popupFrame =
-      nsLayoutUtils::GetPopupFrameForPoint(aRootPresContext, aRootWidget, aPt);
+  nsIFrame* popupFrame = nsLayoutUtils::GetPopupFrameForPoint(
+      aRootPresContext, aRootWidget, aPt,
+      nsLayoutUtils::GetPopupFrameForPointFlags::OnlyReturnFramesWithWidgets);
   return popupFrame ? popupFrame->GetView() : nullptr;
 }
 